.github/workflows/continuous-integration.yml

```diff
@@ -60,47 +66,38 @@ jobs:
           coverage: "xdebug"
           ini-values: "zend.assertions=1"
 
-      - name: "Cache dependencies installed with composer"
-        uses: "actions/cache@v2"
-        with:
-          path: "~/.composer/cache"
-          key: "php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock') }}"
-          restore-keys: "php-${{ matrix.php-version }}-composer-locked-"
-
       # to be removed when doctrine/orm adds support for PHP 8
       - name: "Pretend this is PHP 7.4"
         run: "composer config platform.php 7.4.99"
         if: "${{ matrix.php-version == '8.0' }}"
 
-      - name: "Install dependencies with composer"
-        run: "composer update --no-interaction --prefer-dist"
-        if: "${{ matrix.deps == 'normal' }}"
+      - name: "Use dev stability"
+        run: "composer config minimum-stability dev"
+        if: "${{ matrix.stability == 'dev' }}"
 
-      - name: "Install lowest possible dependencies with composer"
-        run: "composer update --no-interaction --prefer-dist --prefer-lowest"
-        if: "${{ matrix.deps == 'low' }}"
+      - name: "Add extra constraint"
+        run: "composer require --no-update ${{matrix.extra_constraint}}"
+        if: "${{ contains(matrix.extra_constraint, '/') }}"
 
-      - name: "Install dependencies at dev stability with composer"
-        run: "composer config minimum-stability dev; composer update --no-interaction --prefer-dist --prefer-lowest"
```

Comment:
Ugh that was one of my silly mistakes… using `--prefer-lowest` on dev stability :facepalm:

Reply:
That doesn't need to break something in a project. :smile: